service/matching/user_data_manager.go

```diff
@@ -151,6 +151,8 @@ func (m *userDataManagerImpl) WaitUntilInitialized(ctx context.Context) error {
 
 func (m *userDataManagerImpl) Stop() {
 	m.goroGroup.Cancel()
+	// ensure fetch/loadUserData are not running so we can safely call setUserDataState
+	m.goroGroup.Wait()
```

Comment:
would this have to wait for the long poll to expire?

Reply:
Yes, but it's canceling the context that the long poll is using, so it should return quickly.

The alternative to this is using SetIfNotReady in setUserDataState instead of Ready/Set. Maybe we should do that too. But if we don't do the Wait here, that unit test and maybe others will need to reach in and call `m.goroGroup.Wait()` themselves to avoid issues with stuff still happening after the test completes.

Reply:
I think my concern is elevated with the context canceling bit.